res/controllers/midi-components-0.0.js

```diff
@@ -670,6 +676,26 @@
                 });
             }
         },
+        setLayer(newLayer, reconnectComponents) {
+            if (reconnectComponents !== false) {
+                reconnectComponents = true;
+            }
+            if (reconnectComponents === true) {
+                this.forEachComponent(function(component) {
+                    component.disconnect();
+                });
+            }
+
+            Object.assign(this, newLayer);
+
+            if (reconnectComponents === true) {
+                this.forEachComponent(function(component) {
+                    component.connect();
+                    component.trigger();
+                });
+            }
+
+        },
```

Comment:
I decided to create a compy with the new behavior so that it is more explicit that the implementation changes.